idris.nix

```diff
@@ -0,0 +1,22 @@
+{ nixpkgs ? import <nixpkgs> {}}:
+with nixpkgs;
+with pkgs;
+haskell.lib.overrideCabal (haskellPackages.callCabal2nix "idris" ./. {}) (orig:{
+    configureFlags = [ "-fFFI" "-fGMP" ];
+    librarySystemDepends = orig.librarySystemDepends
+      ++ lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [
+        Cocoa
+        CoreServices
+      ]);
+    testHaskellDepends = orig.testHaskellDepends ++ [ nodejs ];
+    preBuild = ''
+      export DYLD_LIBRARY_PATH=`pwd`/dist/build
+    '';
+    preCheck = ''
+      export DYLD_LIBRARY_PATH=`pwd`/dist/build
+      export TARGET=`pwd`
+    '';
+    src = if lib.inNixShell then null else orig.src;
```

Comment:
Haskell packages have an `env` attribute that is appropriate for nix-shell environments, have you tried that?

Reply:
I'm not sure I understand. Do you mean the `src = ...` part is redundant?

Reply:
I mean rather than doing `src = ...` here you can do something like `let pkg = nixpkgs.pkgs.haskell.packages.${compiler}.callPackage ./idris.nix { }; in if inNixShell then pkg.env else pkg`. See the output of `cabal2nix --shell` for an example.

Reply:
OK, I'll get rid of shell.nix and use that instead to avoid polluting the base directory.